src/routers/project/projects.py

```diff
@@ -1,70 +1,139 @@
-from fastapi import APIRouter, HTTPException, Depends
+from fastapi import APIRouter, HTTPException, Depends, Request
 from starlette import status
 
 from sqlalchemy.orm import Session
 from src.project_handler_factory import createHandler
-from src.routers.project.schemas import NewProject, UpdateProject, Project
+from src.routers.project.schemas import NewProject, UpdateProject, Project, InviteProject
 from src.dependecies import get_db
 
 project_router = APIRouter()
 
 @project_router.get("/projects", response_model=list[Project])
-async def get_all_projects(db: Session = Depends(get_db),
+async def get_all_projects(request: Request,
+                           db: Session = Depends(get_db),
                            project_handler: object = Depends(createHandler)):
-    try:    
-        return project_handler.get_all(db)
+    owned = request.headers["owned"]
+    participating = request.headers["participating"]
+    accessible = []
+    for li in [owned, participating]:
+        if len(li.rstrip()) != 0:
+            accessible = accessible + li.split(" ")
+    try:
+        return project_handler.get_all(db, accessible)
     except HTTPException as ex:

```

Comment:
Btw just for the note. Let's assume you have 10k projects for a specific user. You can't simply return this amount of projects in one response.
For such cases, query parameters for the request are used.
limit and offset
You don't need to implement it right now, it's just for the information)

Reply:
I looked up some info about the parameters you mention, understood what they mean and will keep them in mind for future work. Thanks for the note :)